modules/optimization/src/vectorpostprocessors/OptimizationFunctionInnerProductHelper.C

```diff
@@ -102,7 +102,7 @@ OptimizationFunctionInnerProductHelper::getVector(std::vector<Real> & result)
       it.second.resize(nvar);
     result.assign(nvar, 0.0);
 
-    // Integrate in time using quadrature rule
+    // Integrate in time
```

Comment:
The main reason it's a map is so that if this object is called twice on the same time step it overwrites the data instead of appending a vector. So no, it still needs to be a map.